src/Compilers/VisualBasic/Portable/Binding/Binder_Conversions.vb

```diff
@@ -1563,6 +1570,76 @@ DoneWithDiagnostics:
 
         End Function
 
+        Private Function ReclassifyTupleLiteral(
+                       convKind As ConversionKind,
+                       tree As VisualBasicSyntaxNode,
+                       isExplicit As Boolean,
+                       sourceTuple As BoundTupleLiteral,
+                       destination As TypeSymbol,
+                       diagnostics As DiagnosticBag) As BoundExpression
+
+            ' We have a successful tuple conversion rather than producing a separate conversion node 
+            ' which is a conversion on top of a tuple literal, tuple conversion is an element-wise conversion of arguments.
+            Dim isNullableTupleConversion = (convKind = ConversionKind.WideningNullable) Or (convKind = ConversionKind.NarrowingNullable)
+            Debug.Assert(Not isNullableTupleConversion OrElse destination.IsNullableType())
+
+            Dim destinationWithoutNullable = destination
+
+            If isNullableTupleConversion Then
+                destinationWithoutNullable = destination.GetNullableUnderlyingType()
+            End If
+
+            Dim targetType As NamedTypeSymbol = DirectCast(destinationWithoutNullable, NamedTypeSymbol)
+            If targetType.IsTupleType Then
+                Dim destTupleType = DirectCast(targetType, TupleTypeSymbol)
+                ' do not lose the original element names in the literal if different from names in the target
+                ' Come back to this, what about locations? (https:'github.com/dotnet/roslyn/issues/11013)
+                targetType = destTupleType.WithElementNames(sourceTuple.ArgumentNamesOpt)
```

Comment:
Do we need to check that the destination has the same tuple cardinality? Asking because it looks like we do so below.